projects/sandbox/src/app/upload/upload.component.ts

```diff
@@ -63,16 +61,32 @@ export class UploadComponent extends DataPollingComponent {
   rebuildForm(): void {
     this.protocolFields.clearFileValue();
     this.xslFileField.clearFileValue();
-    this.buildForm();
+    this.form.reset();
+    this.error = undefined;
   }
 
+  form = this.fb.group({
+    name: ['', [Validators.required, this.validateDatasetName]],
+    country: ['', [Validators.required]],
+    language: ['', [Validators.required]],
+    uploadProtocol: [ProtocolType.ZIP_UPLOAD, [Validators.required]],
+    url: ['', [Validators.required]],
+    dataset: ['', [Validators.required]],
+    harvestUrl: ['', [Validators.required]],
+    setSpec: [''],
+    metadataFormat: [''],
+    sendXSLT: [''],
+    xsltFile: ['']
+  });
+
   /**
```

Comment:
the above declaration replaces the constructor. That then subscribes listens to form changes ... So you can remove this